src/display/svg.js

```diff
@@ -431,23 +434,35 @@ SVGGraphics = (function SVGGraphicsClosure() {
 
   SVGGraphics.prototype = {
     save: function SVGGraphics_save() {
-      this.transformStack.push(this.transformMatrix);
       var old = this.current;
+      old.parentGroup = this.parentGroup;
+      old.group = this.group;
       this.extraStack.push(old);
       this.current = old.clone();
+      this.current.activeClipUrl = null;
     },
 
     restore: function SVGGraphics_restore() {
-      this.transformMatrix = this.transformStack.pop();
       this.current = this.extraStack.pop();
 
       this.pendingClip = null;
-      this.tgrp = null;
+      this.parentGroup = this.current.parentGroup;
+      this.group = this.current.group;
     },
 
-    group: function SVGGraphics_group(items) {
+    createGroup: function SVGGraphics_group(items) {
       this.save();
+      this.parentGroup = this.group;
+      this.group = this.svgFactory.createElement('svg:g');
+      if (this.current.activeClipUrl) {
+        this.group.setAttributeNS(null, 'clip-path',
+                                  this.current.activeClipUrl);
+      }
+      this.parentGroup.appendChild(this.group);
       this.executeOpTree(items);
+      if (!this.group.children.length) {
```

Comment:
Put a comment above this: `// Remove the group if it is empty after executing the operator tree.`